### ISSUE TITLE
Custom Navigator does not initialize sometimes

### ISSUE THREAD

**Issue by svchostdotexe** [openseadragon/openseadragon#2349]
I have a custom navigator that should render at some div - navigatorId: 'navigatorDiv'. Everytime i route to this component the navigator is not initialized (empty div in DOM) and only appears after a reload. I have tried to debug this for hours to no success, i know this is pretty vague but any help would be much appreciated. Can provide additional information if needed, not sure if the code helps since it is so little.

Using OSD in an angular project.

```ts
const SEADRAGON_OPTIONS: Options = {
      element: this.dragonOutlet.nativeElement,
      zoomPerScroll: 2,
      // hide controls
      showFlipControl: false,
      showHomeControl: false,
      showFullPageControl: false,
      showRotationControl: false,
      showZoomControl: false,
      showNavigator: true,
      navigatorId: 'navigatorDiv',
      navigatorBackground: '#ffffff',
      navigatorBorderColor: '#000000',
      navigatorDisplayRegionColor: '#00FFFF',
      navigatorOpacity: 1,
    };

    this.viewer = this.ngZone.runOutsideAngular(() =>
      OpenSeadragon(SEADRAGON_OPTIONS)
    );
```

**Comment by pearcetm** [comment]
Does this work in previous versions? I.e. 3.1 or before? Does `#navigatorDiv` exist at the time the viewer is created? I'd recommend setting a break point at the beginning of the `Navigator` constructor and stepping through to see where the issue might be.

**Comment by svchostdotexe** [comment]
#navigatorDiv exists at viewer creation, already checked that. 
Downgraded to 3.1 to check, still not working either.

Will try the breakpoint!

**Comment by svchostdotexe** [comment]
just ran it through the debugger multiple times and cant find where the issue is, but will continue.

something i just noticed: adding an explicit width and height to the navigator config makes it work

```
navigatorId: 'navigatorDiv',
navigatorWidth: 200,
navigatorHeight: 200,```

but the div loaded by "getElementById" has the correct size in the debugger, so not sure how this solves the issue

**Comment by svchostdotexe** [comment]
well, i found the solution! i had a navigation in between two viewer creations and did not know that you had to clean up the viewer with `viewer.destroy()`. because of that the first viewer interfered with the creation of the second one. hope this helps someone in the future. even though i worked a lot with OSD until now this wasn't obvious to me, a hint somewhere in the tutorials and how to's might help others in the future :)

**Comment by pearcetm** [comment]
Glad you found the root of the problem! I'm not an angular user so I really don't know how that framework may contribute to this issue. Out of curiosity, are you following the recommendation from the [online docs regarding angular](http://openseadragon.github.io/docs/)?

> Important note for Angular users
By default Angular enables change detection in all imported code, which introduces a lot of CPU load in UI intensive libraries. To avoid this, you must create Openseadragon object out of [NgZone](https://angular.io/guide/zone). Instead of
```
const viewer = OpenSeadragon(options);
```
> use
```
const viewer = this.ngZone.runOutsideAngular(() =>
  OpenSeadragon(options)
);
```
> More info in the [Angular documentation](https://angular.io/guide/zone#ngzone-run-and-runoutsideofangular).

**Comment by svchostdotexe** [comment]
yes, we are creating the viewer exactly like this! Not sure if angular contributes to this issue or the browser somehow still had some remnants of the first viewer creation and therefore failed to create the second one. happy that it is resolved though, that was a tough one, especially since i was searching for the mistake in the wrong place for so long!

**Comment by iangilman** [comment]
@svchostdotexe Thank you for sharing your solution! I'll tag this for documentation so someone can add something about destroying the viewer if you want to reuse the space. Any thoughts about where might be a good place to put that info?

**Comment by svchostdotexe** [comment]
since i had the problem with the navigator maybe directly on the navigator page?
https://openseadragon.github.io/examples/ui-viewport-navigator/
even though it does not fit there directly, a little hint on the bottom might save others a headache!

yes we reuse the viewer usually, but in this case there is a navigation in between, so we just reconstruct everything. when i have more time on my hands reusing the old viewer would be a nice performance increase.
